### PR TITLE
RM-79371 Release over_react 3.8.0

### PR DESCRIPTION
**Pull request by aaronlademann-wf**
The latest release of over_react includes...

__New Features__

- 🎉 🎉 🎉 __Support for function components, memo and hooks!!!__ 🎉 🎉 🎉

    Sooooo much work from so many amazing people made this possible, but to summarize: 
    
    - #606 Add support for function components
    - #613 Add support for `memo` higher order component
    - #611 Hooks, hooks, and more hooks!
        - useState
        - useCallback
        - useContext
        - useEffect
        - useLayoutEffect
        - useReducer
        - useRef
        - useMemo
        - useImperativeHandle
        - useDebugValue

    <p><br>It works like this...</p>
    
    Define the component
    ```dart
    mixin FancyBorderProps on UiProps {
      String color;
    }
    
    UiFactory<FancyBorderProps> FancyBorder = uiFunction(
      (props) {
        // props is typed as a `FancyBorderProps`
        // whatever you return here will be rendered
        return (Dom.div()..className = 'fancy-border border-${props.color}')( 
          props.children,
        );
      }, 
      // this is the `UiFactoryConfig`
      $FancyBorderConfig, // ignore: undefined_identifier
    );
    ```
    
    Render the component _(exact same consumer API as a class-based component)_:
    ```dart
    import 'package:over_react/over_react.dart';
    import 'fancy_border.dart'; // Where your component is defined
    
    main() {
      final renderedWidget = (FancyBorder()..color = /* some color value */)(
        // put some children here!
      );
    
      react_dom.render(renderedWidget, querySelector('#idOfSomeNodeInTheDom'));
    }
    ```
  
__Other Changes__

- #612 Deprecate `forwardRef` / add `uiForwardRef` as its replacement

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,67 @@
...
+    
+    UiFactory<FancyBorderProps> FancyBorder = uiFunction(
+      (props) {
+        // props is typed as a `FancyBorderProps`
+        // whatever you return here will be rendered
+        return (Dom.div()..className = 'fancy-border border-${props.color}')( 
+          props.children,
+        );
+      }, 
+      // this is the `UiFactoryConfig`
```
Comment (greglittlefield-wf) [inline CHANGELOG.md:41]: #nit Saying what type this is isn't helpful in this context
```suggestion
```

File: pubspec.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: over_react
-version: 3.7.0
+version: 3.8.0
```
Comment (greglittlefield-wf) [inline pubspec.yaml:2]: Don't forget to bump in the analyzer plugin pubspec

**Comment by aaronlademann-wf** [comment]
+1 to https://github.com/Workiva/over_react/pull/616/commits/cb1ab3cc537d799f94e896515c522fa8be62b5dc

**Comment by aaronlademann-wf** [comment]
@Workiva/release-management-pp

**Pull request merged**